tests/test_grizzly/users/test_iothub.py

```diff
@@ -0,0 +1,193 @@
+    scenario.context['host'] = 'test'
+
+    request.method = RequestMethod.SEND
+
+    scenario.tasks.add(request)
+
+    return cast(IotHubUser, user), scenario, environment
+
+
+class TestIotHubUser:
+    @pytest.mark.usefixtures('iot_hub_scenario')
+    def test_create(self, iot_hub_scenario: IoTHubScenarioFixture) -> None:
+        user, _, environment = iot_hub_scenario
+        assert issubclass(user.__class__, GrizzlyUser)
+
+        print('1')
+        IotHubUser.host = 'PostName=my_iot_host_name;DeviceId=my_device;SharedAccessKey=xxxyyyyzzz=='
+        with pytest.raises(ValueError) as e:
+            user = IotHubUser(environment)
+        assert 'host needs to start with "HostName="' in str(e)
+
+        print('2')
+        IotHubUser.host = 'HostName=my_iot_host_name'
+        with pytest.raises(ValueError) as e:
+            user = IotHubUser(environment)
+        assert 'needs DeviceId and SharedAccessKey in the query string' in str(e)
+
+        print('3')
+        IotHubUser.host = 'HostName=my_iot_host_name;SharedAccessKey=xxxyyyyzzz=='
+        with pytest.raises(ValueError) as e:
+            user = IotHubUser(environment)
+        assert 'needs DeviceId in the query string' in str(e)
+
+        print('4')
+        IotHubUser.host = 'HostName=my_iot_host_name;DeviceId=my_device'
+        with pytest.raises(ValueError) as e:
+            user = IotHubUser(environment)
+        assert 'needs SharedAccessKey in the query string' in str(e)
+
+    @pytest.mark.usefixtures('iot_hub_scenario')
+    def test_send(self, iot_hub_scenario: IoTHubScenarioFixture, mocker: MockerFixture, grizzly_fixture: GrizzlyFixture) -> None:
+        [user, scenario, _] = iot_hub_scenario
+
+        grizzly = grizzly_fixture.grizzly
+
+        remote_variables = {
+            'variables': transform(grizzly, {
+                'AtomicIntegerIncrementer.messageID': 31337,
+                'AtomicDate.now': '',
+                'messageID': 137,
+            }),
+        }
+        user.add_context(remote_variables)
+        request = cast(RequestTask, scenario.tasks[-1])
+        request.endpoint = 'not_relevant'
+
+        upload_blob = mocker.patch('azure.storage.blob._blob_client.BlobClient.upload_blob', autospec=True)
+        notify_blob_upload_status = mocker.patch('azure.iot.device.IoTHubDeviceClient.notify_blob_upload_status', autospec=True)
+
+        expected_payload = {
+            'result': {
+                'id': 'ID-31337',
+                'date': '',
+                'variable': '137',
+                'item': {
+                    'description': 'this is just a description',
+                }
+            }
+        }
+
+        metadata, payload = user.request(request)
+
+        assert payload is not None
+        assert upload_blob.call_count == 1
+        args, _ = upload_blob.call_args_list[-1]
+        assert len(args) == 2
+        assert args[1] == json.dumps(expected_payload, indent=4)
```

Comment:
hm, maybe do it the other way around, `json.loads(args[1])` and we don't have to care about indentation.